internal/datastore/mysql/readwrite.go

```diff
@@ -37,6 +39,22 @@ type mysqlReadWriteTXN struct {
 	newTxnID uint64
 }
 
+// caveatContextWrapper is used to marshall maps into MySQLs JSON data type
+type caveatContextWrapper map[string]any
+
+func (cc *caveatContextWrapper) Scan(val any) error {
+	switch v := val.(type) {
```

Comment:
why a type switch and not just a type assertion? i.e. `v, ok := val.([]byte)`

Reply:
can do!